---
title: "Assignment 3 R Notebook"
output:
  html_document:
    fig_caption: true
    toc: true
    toc_depth: 2
bibliography: references2.bib
---

```{r message=FALSE}
if (!require("dplyr", quietly = TRUE))
  install.packages("dplyr")
library("dplyr")
```

```{r test-main, child = './A2_DienNguyen.Rmd'}
```

# Summary of previous analyses
The data set under investigation is from a GEO (Gene Expression Omnibus) 
[accession](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE144474)
titled "Contribution of brain pericytes in blood-brain barrier (BBB)
formation and maintenance: A transcriptomic study of cocultured human 
endothelial cells derived from hematopoietic stem cells" [@data_paper]. 

The interaction of endothelial cells (ECs) and brain pericytes induces BBB 
characteristics in brain ECs during embryogenesis. Pericytes are a type of 
vascular cells embedded in the basement membrane, and is important for BBB 
functioning, so they can be used to differentiate stem cells into ECs and them 
BBB in vitro. However, the molecular events in BBB maturation are not fully 
understood.

The data was first downloaded from GEO. Then, simple statistics like finding the mean and standard deviation, and generating plots of gene counts, were done to assess data quality. Each gene was mapped to a HUGO gene symbol, from an ENTREZ gene symbol. Low quality reads were removed in the data cleaning step. The, normalization was applied. 

Differential expression analysis was carried out to see which genes are expressed at high or low amounts. 753 genes passed the p-value threshold of 0.05, but only 7 passed FDR corrrection. There were a total of 185 differentially expressed genes.

Thresholded over expression analysis (ORA) was carried out. The thresholds are +0.7 and -0.7 logFC. ORA was run using gprofiler, with the February release of the GO:BP data set. 

# Assignment 3

# Non-thresholded gene set enrichment analysis

To carry out pre-ranked non-thresholded gene set enrichment analysis, a .rnk file which contains genes ranked by their rank score. The rank score considers both the direction of the change and the significance. The genes are then ranked from highest to lowest score. 

## Generate .rnk file
```{r}
# Calculate rank of all genes
qlf_all$table$RankScore <- as.numeric(sign(qlf_all$table$logFC) * (-log10(qlf_all$table$PValue)))

# Rank them from largest to smallest
ranked_genes <- dplyr::arrange(qlf_all$table, desc(RankScore))[c("RankScore")]
Gene <- rownames(ranked_genes)
ranked_genes <- cbind(Gene, ranked_genes)
write.table(ranked_genes, file = "edgeR_results_2.rnk", sep = "\t", row.names = FALSE, quote = FALSE)
```

Running gene enrichment analysis also requires a file containing gene sets for reference. Here, we are using the Bader Lab's March 01 2024 release of human enrichment map gene sets with no PFOCR and no IEA [@enrichmentmap_ref]. The files are downloaded directly from the website.

## Run GSEA on desktop app

Non-thresholded enrichment analysis was done using the GSEA desktop application [@gsea_ref]. The following figures show the results of the enrichment analysis. 

![Figure 1. The top 6 up-regulated processes in the cocultured sample, generated by GSEA, which uses is a non-thresholded gene set enrichment analysis method. It was also performed with a pre-ranked gene list. The black vertical bars below each graph show the gene that is enriched at s pecific level on the red-blue scale [@gsea_ref].](figures/GSEA_up_snapshot.png)

![Figure 2. The top 6 down-regulated processes in the cocultured sample, generated by GSEA, which uses is a non-thresholded gene set enrichment analysis method. It was also performed with a pre-ranked gene list [@gsea_ref].](figures/GSEA_down_snapshot.png)


## Report questions
1. The method used for gene enrichment analysis is GSEA, using the desktop application. The geneset used was from the Bader Lab's March 01 2024 release of human enrichment map gene sets. I chose the GOBP, no PFOCR, no IEA option to get more definite results.
2. There are 1694 out of 5628 gene sets that are upregulated, of which only 207 passed correction of FDR < 25%. The top up-regulated processes are post-translational protein phosphorylation, regulation of IGF activity by IGFBP, and IL4-mediated signalling events. There are also other signalling pathways that are upregulated, and some growth processes like collagen formation and protein-lipid complex organization. There are 3834 gene sets that are down-regulated, of which 619 passed correction. The top down-regulated processes are mitochondrial translation termination and dectin-1 mediated noncanonical NF-KB signalling. 

3. One of the similiar up-regulated pathways found in both gprofiler and GSEA is regulation of IGF activity by IGFBP. Another pathway that is upregulated in both programs is ADP metabolic process. However in gprofiler, not many signalling pathways were shown to be up-regulated. For the down-regulated processes, the results were very different. In gprofiler, most of the processes were very broad, some at organism-level process. In GSEA, the processes were very specific, like the regulation of specific proteins and signalling molecules. Since the results are on different levels of specificity, they are not straight-forward to compare.

# Enrichment map

## Initial map and Report question 1

An enrichment map was generated using Cytoscape, with the GSEA results as input. The enrichment map contains 361 nodes and 7823 edges. The FDR cutoff value was set to be to be 0.1, but by default it is set to 0.0001. When the enrichment map analysis was run using the default FDR cut-off, very few red nodes were generated. Everything else was set according to the default settings. p-value cutoff is 0.1 and NES is set to All. For the number of edges, data set edges were set to automatic, and connectivity is set to medium. The metric for calculating edges is Jaccard+Overlap combined, with cutoff set to 0.375.


![Figure 3. Enrichment map generated by Cytoscape using the results from GSEA [@cytoscape_ref]. Each red notes indicates an up-regulated gene set and each blue node indicates a down-regulated gene set. Each edge between two nodes indicate that the gene sets share at least one gene. Most of the nodes in this figure are blue, since many of the up-regulated processes don't pass the p-value threshold. This enrichment map has not been manually adjusted for readability.](figures/EM_initial.png)


## Annotated map and Report question 2

The enrichment map was annotated using AutoAnnotate, one of Cytoscape's extension tool. These were the parameters used for the annotation: The annotation was created using clusterMaker app, with a cluster algorithm called MCL Cluster.The label column was assigned GS_DESCR, which is the description column of the processes. The label algorithm chosen was WordCould: Adjacent Words. The max words per label was 3, min word occurence is 1, and adjacent word bonus is 8. The "Label network to prevent overlap" option was also selected to make it easier to interpret. 

![Figure 4. Annotated enrichment map. The original enrichment map was annotated using AutoAnnotate [@autoannotate_ref]. The nodes are clustered and each clustered has an overall common theme.](figures/EM_annotated_with_legend.png)

## Theme network and Report question 3

The annotated enrichment map was collapsed into a theme network, using AutoAnnotate's "Create Summary Network" feature. This creates a map of the main themes of the original enrichment map. Some major themes of the down-regulated processes include proteosome degradation apc, apoptotic mitochondrial mitochondrion and splicing transesterification. One major theme of the up-regualted processes is ribonucleoside diphosphate metabolic. Most up-regulated processes don't cluster, and exist as separate nodes. 

![Figure 4. A theme network generated by Cytoscape, which is a summary network of the annotated enrichment map [@cytoscape_ref].](figures/summary_network.png)

# Interpret results
In the original paper, processes like cell-cell junction tightening and ATP metabolism were shown to be up-regulated. It was also suspected that the Wnt signalling pathways is involved in communications between epithelial cells and brain pericytes. Similar results were seen in our enrichment analysis. For instance, some of the up-regulated processes are protein-lipid complex organization and collagen formation. The ADP metabolic process is also up-regulated, similar to the original paper. However, there were no signs of the Wnt pathway being up-regulated.

Compared to Assignment 2, the top up-regulated processes are similar. However, some highly expressed genes like NOTCH3, which is in involved in the Notch signalling pathway, are not significant in the GSEA analysis. In terms of the down-regulated processes, the ones in Assignment 2, produced by gprofiler, are too broad, whereas the ones produced by GSEA are very specific. Since they are so small, it is hard to pinpoint exactly which bigger processes they play a role in. 

The other top up-regulated processes are mostly signalling pathways, or related. Post-translational protein phosphorylation plays an important role in cell signalling, and can be involved in the main pathways that induce brain-blood barrier formation[@protein_phospho_ref]. Many IGFBP genes are highly expressed in the co-cultured samples, so one of the top up-regulated processes is the regulation of IGF (insulin-like growth factor) activity by IGFBP. Some IGF activity, like IGF1, have been shown to be crucial in normal brain development [@igf_ref].

Further analysis needs to be done on how IGF activity, the IL4-mediated signalling pathway, and protein phosphorylation are involved in development of the blood-brain barrier, specifically how each molecule in the pathways interact with each other. 


# Visualize pathway using GeneMANIA

The PID-IL4 pathway (IL4 mediated signalling events) was selected for further analysis using a network visualization tool called GeneMANIA. This pathway was selected because it is one of the pathways of interest in the original paper. Studying the genes involved in this pathway will help elucidate the molecular mechanisms involved in the development of the blood-brain barrier, since they are still unknown.

```{r}
pathway_genes <- read.csv("GeneMANIA_network_genes.csv")$gene.name

# Get the pvalues and logFC change of those genes
qlf_all$table[rownames(qlf_all$table) %in% pathway_genes,]

```

![Network of genes in IL4-mediated signalling pathway. The color of each edge indicates the type of interaction between the two genes (nodes) that share the edge. The network is generated by GeneMANIA [@genemania_ref]. Each node is annotated with the lof fold change and p-value of their expression.](figures/GeneMANIA_annotated.png)

# References


